pkg/scheduler/scheduler.go

```diff
@@ -845,11 +885,7 @@ func (ks *KubernetesScheduler) recoverTasks() error {
 		defer ks.Unlock()
```

Comment:
I actually planned to remove the redundant locks in a second round. Is that ok or do you prefer this in the same patch?